extensions/mysql/index.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+const Promise = require('bluebird');
+const mysql = require('mysql');
+const crypto = require('crypto');
+const omit = require('lodash/omit');
+const cli = require('../../lib');
+
+class MySQLExtension extends cli.Extension {
+    setup(cmd, argv) {
+        // ghost setup --local, skip
+        if (argv.local) {
+            return;
+        }
+
+        cmd.addStage('mysql', this.setupMySQL.bind(this));
+    }
+
+    setupMySQL(argv, ctx, task) {
+        let self = this;
+        this.databaseConfig = ctx.instance.config.get('database');
+
+        return this.canConnect()
+            .then(function () {
+                if (self.databaseConfig.connection.user === 'root') {
+                    return self.ui.confirm('Your MySQL user is root. Would you like to create a `ghost` MySQL user?', true)
+                        .then(function (res) {
+                            if (res.yes) {
+                                return self.createMySQLUser(ctx);
+                            }
+                        });
+                }
+
+                self.ui.log('MySQL: Your user is: ' + self.databaseConfig.connection.user, 'green');
+            })
+            .finally(function () {
+                self.connection.end();
+            });
+    }
+
+    canConnect() {
+        let self = this;
+        this.connection = mysql.createConnection(omit(self.databaseConfig.connection, 'database'));
+
+        return new Promise(function (resolve, reject) {
+            self.connection.connect(function (err) {
+                if (err) {
+                    self.ui.log('MySQL: connection error.', 'yellow');
+                    return reject(new cli.errors.SystemError(err.message));
+                }
+
+                self.ui.log('MySQL: connection successful.', 'green');
+                resolve();
+            });
+        });
+    }
+
+    createMySQLUser(ctx) {
+        let self = this;
+        let randomPassword = crypto.randomBytes(10).toString('hex');
+        let host = self.databaseConfig.connection.host;
+
+        return new Promise(function (resolve, reject) {
```

Comment:
Instead of the `return new Promise((resolve, reject) => {})` usage - bluebird has a `Promise.fromCallback` method, would probably be good to abstract that out into a helper method in this class.

e.g.
```js
_query(queryString) {
  return Promise.fromCallback(cb => this.connection.query(queryString, cb));
}
```